inc/integrations/google/auth/google-auth/google-auth.php

```diff
@@ -47,12 +64,38 @@ public static function generate_token_from_service_account_key(
 			return $service_account_key;
 		}
 
+		$cache_key = 'google_auth_token_' . $service_account_key->client_email;
+		if ( ! $no_cache ) {
+			$cached_token = wp_cache_get( $cache_key, 'oauth-tokens' );
+			if ( false !== $cached_token ) {
+				return $cached_token;
+			}
+		}
+
 		$jwt       = self::generate_jwt( $service_account_key, $scope );
 		$token_uri = $service_account_key->token_uri;
 
-		return self::get_token_using_jwt( $jwt, $token_uri );
+		$token = self::get_token_using_jwt( $jwt, $token_uri );
+
+		if ( ! is_wp_error( $token ) ) {
+			wp_cache_set(
+				$cache_key,
+				$token,
+				'oauth-tokens',
+				3000, // 50 minutes
+			);
```

Comment:
@chriszarate @mhsdef I am wondering if we should encrypt this token before writing to cache ? We do encrypt the data source config when we store it in options table, but that is persistent as opposed to this which has TTL of 50mins for now.

Reply:
Ideally, yeah. We could do what we're doing with the DB option if there's a hook that fires in the cache set. I didn't see one, in brief perusing, but maybe I missed it.

Otherwise, could invoke our encryption code manually to handle.